client/proposal.go

```diff
@@ -264,6 +266,9 @@ func (c *Client) proposeTwoPartyChannel(
 
 	env, err := receiver.Next(ctx)
 	if err != nil {
+		if pcontext.IsContextError(err) {
+			return nil, newRequestTimedOutError("channel proposal", err.Error())
```

Comment:
Wouldn't it be better to wrap the error object instead of just the error string?

Reply:
I wrap the `newRequestTimedOut` error instead of the actual error because it is this error that the user of the API will be checking for. If we do it otherwise: wrap the actual error and add the `newRequestTimedOutError` as a message, I think it would not be possible for to check for `RequestTimedOutError`.

Reply:
I add the error string, so that the details of the original error message is still retained.

Reply:
@matthiasgeihs However, if the underlying error is to be retained along with this error. Then one option is to store the actual error in the `RequestTimeOutError` struct and write an `Unwrap` method for it (see [here](https://play.golang.org/p/iJKrTEQApuq)) which will return the underlying error.